### ISSUE TITLE
0.8 meta issue

### ISSUE THREAD

**Issue by bmcfee** [mir-evaluation/mir_eval#369]
:wave: Seems like we're overdue for a new release here, if for no reason other than #352 (mir_eval currently doesn't work on the most recent numpy).

I've gone ahead and tagged a few issues that seemed particularly important / relevant to include here, and am happy to take the lead on knocking some of this out.  In particular, I'd like to modernize the CI / test / packaging environments.

@craffel any thoughts on this?  I know it's been a while, but I think we should aim for a quick turnaround here, and I have a few spare cycles right now to get things moving.

**Comment by craffel** [comment]
Yes, happy to have a new release! I'm on parental leave, so happy to have you do it if you have spare cycles. Thanks.

**Comment by bmcfee** [comment]
Congrats!  Seems like there's a lot of that going around these days :grin:  (my leave is just winding down now, hence the spare cycles before everyone else notices...)

I can manage most of this myself, but we will need to coordinate a little on setting up the release.  The best practices now using GH actions to automate pypi push on release make it pretty easy to deploy, but you'll need to do a little configuring on the mir_eval pypi settings to enable it.

See https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

I just set this up in pescador yesterday, so we can go off that as a template: https://github.com/pescadores/pescador/blob/main/.github/workflows/publish.yml

**Comment by craffel** [comment]
Great - just bug me with the steps I need to take. There may be some latency (right now the baby is napping so latency is low).

**Comment by bmcfee** [comment]
Release candidate made its way out - I had to modernize some workflow stuff for the pypi upload, but it seems to all be working now.  I'll leave it in rc status for a day or so, but if nobody finds any glaring problems, we can finally get this off the stack!

Last step upon release will be to set up a redirect on the gh-pages to point to the new rtd build.
